packages/ui/src/kitchen-sink-manager.tsx

```diff
@@ -114,28 +122,43 @@ export class KitchenSinkManager {
     };
 
     const relevantPanel = this._panels[data.position];
-    const item = new KitchenSinkItem(itemOptions);
-    relevantPanel.items.push(item);
-    if (relevantPanel.ref) {
-      relevantPanel.ref.update();
-    }
-
-    this.options.upperBarManager.add({
+    const kitchenSinkItem = new KitchenSinkItem(itemOptions);
+    const upperBarItem = this.options.upperBarManager.add({
       label: data.label,
       renderItem: data.renderIcon,
-      onClick: () => this._toggle(item),
+      onClick: () => this._toggle(kitchenSinkItem),
+    });
+    relevantPanel.items.push({
+      kitchenSinkItem,
+      upperBarItem,
     });
+    if (relevantPanel.ref) {
+      relevantPanel.ref.update();
+    }
 
-    return item;
+    return kitchenSinkItem;
   }
 
-  //todo [sa] implement
   remove(item: KitchenSinkItem): void {
-    //deactivate item if needed
-    //remove upper bar icon
-    //remove from _items
-    //call item.destroy
-    //refresh UI ???
+    const relevantPanel = this._panels[item.data.position];
+    const itemsIndex = ArrayUtils.findIndex(
+      relevantPanel.items,
+      ({kitchenSinkItem}) => {
+        return item === kitchenSinkItem;
+      }
+    );
+
+    if (itemsIndex === -1) return;
+
+    this._deactivateItem(item);
+    const {upperBarItem} = relevantPanel.items[itemsIndex];
+    item.destroy();
```

Comment:
this is internal, probably being called by the upperBarManager. can you confirm? can you move it to there? we need to improve our api of the items but for now wdyt? about changing `destroy` to `public _destroy`?

Reply:
talked about it...